core/src/main/java/org/geysermc/geyser/network/GeyserServerInitializer.java

```diff
@@ -72,6 +80,38 @@ public void initSession(@NonNull BedrockServerSession bedrockServerSession) {
         }
     }
 
+    @Override
+    protected void postInitChannel(Channel channel) throws Exception {
+        super.postInitChannel(channel);
+
+        channel.pipeline().addLast(new SimpleChannelInboundHandler<ByteBuf>() {
+            private static final int allowedExceptions = 5;
+            private static final long delayUntilExceptionsReset = 60;
+
+            private long lastException;
+            private int exceptionCounts;
+
+            @Override
+            protected void channelRead0(ChannelHandlerContext ctx, ByteBuf msg) throws Exception {
+                ctx.fireChannelRead(msg);
+            }
+
+            @Override
+            public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+                final long currentTime = System.currentTimeMillis();
+                if (currentTime >= lastException + (delayUntilExceptionsReset * 1000)) {
+                    exceptionCounts = 0;
+                }
+                lastException = currentTime;
+                if (exceptionCounts++ > allowedExceptions) {
+                    ctx.pipeline().get(GeyserBedrockPeer.class).close("Too many exceptions created.");
```

Comment:
This is a bit vague for a client.. maybe change the reason here to `Sent too many invalid packets.` instead?

Reply:
This will be removed as it doesn't do anythingz